src/Libraries/RevitNodesUI/Selection.cs

```diff
@@ -155,8 +175,11 @@ public override void Dispose()
 
             if (revitDynamoModel != null)
             {
-                var hwm = RevitDynamoModel.Workspaces.OfType<HomeWorkspaceModel>().ElementAt(0);
-                hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
+                var hwm = revitDynamoModel.CurrentWorkspace as HomeWorkspaceModel;
+                if (hwm != null)
+                {
+                    hwm.RunSettings.PropertyChanged -= revMod_PropertyChanged;
```

Comment:
Here is when the selection node is disposed.

The one in the setter is to un-register the old event handler for the old model and register the event handler for the new model.